pkg/config/piped.go

```diff
@@ -969,14 +974,23 @@ func (n *NotificationReceiverSlack) Validate() error {
 	if len(mentionedAccounts) > 0 {
 		n.MentionedAccounts = mentionedAccounts
 	}
-	if n.HookURL != "" && (n.OAuthToken != "" || n.ChannelID != "") {
-		return errors.New("only one of hookURL or oauthToken and channelID should be used")
+	if n.HookURL != "" && (n.OAuthToken != "" || n.OAuthTokenFile != "" || n.OAuthTokenData != "" || n.ChannelID != "") {
+		return errors.New("only one of hookURL or OAuthToken, OAuthTokenData, OAuthTokenFile and channelID should be used")
 	}
 	if n.HookURL != "" {
 		return nil
 	}
-	if n.OAuthToken == "" || n.ChannelID == "" {
-		return errors.New("missing oauthToken or channelID configuration")
+	if (n.HookURL == "" && n.ChannelID == "") || (n.OAuthToken == "" && n.OAuthTokenFile == "" && n.OAuthTokenData == "") {
```

Comment:
At this place, `n.HookURL == ""` has no mean since `n.HookURL` will always be empty here. Isn't it?

Reply:
```suggestion
	if n.ChannelID == "" || (n.OAuthToken == "" && n.OAuthTokenFile == "" && n.OAuthTokenData == "") {
```
I think it should be like this

Reply:
fyi, we need either `HookURL` or `ChannelID & one of the ways to provide token` to make the slack sender work